src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Services/SyntaxFacts/ISyntaxFacts.cs

```diff
@@ -17,6 +17,64 @@
+    /// VB.  These functions should throw if passed a node that the corresponding 'IsYYY' did not return <see langword="true"/> for.
+    /// </item>
+    /// <item>
+    /// 'GetPartsOfXXX(SyntaxNode node, out SyntaxNode/SyntaxToken part1, ...)' where 'XXX' one of the same named Syntax constructs
+    /// that both C# and VB have, and where the returned parts correspond to the members those nodes have in common across the 
+    /// languages.  For example 'GetPartsOfQualifiedName(SyntaxNode node, out SyntaxNode left, out SyntaxToken dotToken, out SyntaxNode right)'
+    /// VB.  These functions should throw if passed a node that the corresponding 'IsXXX' did not return <see langword="true"/> for.
+    /// </item>
+    /// <item>
+    /// Absolutely trivial questions that relate to syntax and can be asked sensibly of each language.  For example,
+    /// if certain constructs (like 'patterns') are supported in that language or not.
+    /// </item>
+    /// </list>
+    /// Importantly avoid:
+    /// <list type="bullet">
+    /// <item>
+    /// Functions that attempt to blur the lines between similar constructs in the same language.  For example, a QualifiedName
+    /// is not the same as a MemberAccessExpression (despite A.B being representable as either depending on context). 
+    /// Features that need to handle both should make it clear that they are doing so, showing that they're doing the right
+    /// thing for the contexts each can arise in (for the above example in 'type' vs 'expression' contexts).
+    /// </item>
+    /// <item>
+    /// Functions which are effectively specific to a single feature are are just trying to find a place to place complex
+    /// feature logic in a place such that it can run over VB or C#.  For example, a function to determine if a position
+    /// is on the 'header' of a node.  a 'header' is a not a well defined syntax concept that can be trivially asked of
+    /// nodes in either language.  It is an excapsulation of a feature (or set of features) level idea that should be in
+    /// its own dedicated service.
+    /// </item>
+    /// <item>
+    /// Functions that mutate or update syntax constructs for example 'WithXXX'.  These should be on SyntaxGenerator or
+    /// some other feature specific service.
+    /// </item>
+    /// <item>
+    /// Functions that a single item when one language may allow for multiple.  For example 'GetIdentifierOfVariableDeclarator'.
+    /// In VB a VariableDeclarator can itself have several names, so calling code must be written to check for that and handle
+    /// it apropriately.  Functions like this make it seem like that doesn't need to be considered, easily allowing for bugs
+    /// to creep in.
+    /// </item>
+    /// </list>
+    /// </summary>
+    /// <remarks>
+    /// Many helpers in this type currently violate the above 'dos' and 'do nots'.  They should be removed and either 
```

Comment:
What's the plan for this? Did you audit functions?

Reply:
> What's the plan for this?

I have no plan currently.  Perhaps over time i (and others) can address deviations.

Reply:
I guess unless we're going to have a plan to address this...I'm not sure what the point is in setting guidance we don't meet and don't ever plan to meet.

Reply:
I'd make an alternate suggestion (and mentioned this to Cyrus privately): let's at least go through the current methods and stick TODO comments for which ones we think are violating but aren't immediately easy to fix in this PR. The challenge with this comment is if I'm somebody new to the codebase, and I read this, but I also see a bunch of violations, I don't know which is wrong. I can see *something* is wrong, but I don't know what.